src/kemal/ext/context.cr

```diff
@@ -7,35 +7,28 @@ class HTTP::Server
     # :nodoc:
     STORE_MAPPINGS = [Nil, String, Int32, Int64, Float64, Bool]
 
-    property! app : Kemal::Base
-
     macro finished
       alias StoreTypes = Union({{ *STORE_MAPPINGS }})
       getter store = {} of String => StoreTypes
     end
 
     def params
-      @request.url_params ||= route_lookup.params
       @params ||= if @request.param_parser
                     @request.param_parser.not_nil!
                   else
                     Kemal::ParamParser.new(@request)
                   end
     end
 
+    def initialize_url_params(app)
```

Comment:
This should not be initialized unless `params` is explicitly called a.k.a lazy evaulate.